### PR TITLE
Fix `axis` of `cupy.gradient`

### LINKED ISSUE #4520: `cupy.gradient` should not sort axis
```
>>> for xp in [numpy, cupy]:
...     print(xp.__name__)
...     for a in xp.gradient(xp.arange(4).reshape(2, 2), axis=(1, 0)):
...         print(a)
...
numpy
[[1. 1.]
 [1. 1.]]
[[2. 2.]
 [2. 2.]]
cupy
[[2. 2.]
 [2. 2.]]
[[1. 1.]
 [1. 1.]]
```

### PR DESCRIPTION
**Pull request by toslunar**
Fix #4520.

Please comment on #4522, too

### CODE REVIEW

**Review by grlee77 — APPROVED** [review]
Good catch, @toslunar!

File: cupy/_math/sumprod.py [hunk 1]
```diff
@@ -341,7 +342,7 @@ def gradient(f, *varargs, axis=None, edge_order=1):
     if axis is None:
         axes = tuple(range(ndim))
     else:
-        axes = internal._normalize_axis_indices(axis, ndim)
+        axes = nx.normalize_axis_tuple(axis, ndim)
```
Comment (asi1024) [inline cupy/_math/sumprod.py:345]: `internal._normalize_axis_indices(axis, ndim, sort_axes=False)` is also fine. Let's discuss we use which one in #4522.

**Comment by toslunar** [comment]
Let's postpone #4522.

**Comment by asi1024** [comment]
Jenkins, test this please.

**Comment by asi1024** [comment]
@toslunar Could you fix for test failure?

**Comment by toslunar** [comment]
Jenkins, test this please.

**Comment by asi1024** [comment]
LGTM.

**Pull request merged**